bundle/config/mutator/trampoline.go

```diff
@@ -0,0 +1,95 @@
+package mutator
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"text/template"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/databricks-sdk-go/service/jobs"
+)
+
+type fnTemplateData func(task *jobs.Task) (map[string]any, error)
+type fnCleanUp func(task *jobs.Task)
+type fnTasks func(b *bundle.Bundle) []*jobs.Task
```

Comment:
Instead of a set of function types, you could also define an interface. That enforces downstream must implement all of them.

```
type TrampolineFunctions interface {
  GetTasks() ...
}
```